### PR TITLE
feat: support editing increaseAllowance/increaseApproval in send confirm(OK-40232)

### PR DESCRIPTION
**Pull request by weatherstar**
## Summary
- Recognize `increaseAllowance` (OZ) and `increaseApproval` (legacy OZ v1, selector `0xd73dd623`) as editable approve actions in the EVM decoder, so users can adjust the delta from the send confirm "Edit allowance" dialog.
- Display the post-tx total (current on-chain allowance + delta) on the confirm page and as Current/Final rows in `ApproveEditor`, fetched via `/swap/v1/allowance`.
- Harden the increase-allowance UI against NaN, cross-token misattribution, and silently swallowed allowance lookup errors.

## Intent & Context
Previously the "Edit allowance" dialog only recognized the canonical `approve(spender, amount)` selector. dApps that call `increaseAllowance` / `increaseApproval` (delta-based, never overwrites the prior allowance) bypassed the editor entirely, leaving users unable to adjust the requested delta and unable to see what their final allowance would actually be after the tx — a UX gap that becomes a safety issue when the dApp asks for a large delta on top of an existing allowance.

OK-40232 tracks closing that gap end-to-end.

## Root Cause
Three orthogonal issues compounded the problem:
1. The decoder mapped only the `approve(...)` selector, so inc/dec calls were not classified as approve actions and never reached the editor.
2. When the editor was wired up, the `Unlimited` toggle would silently switch the encoded call to `approve(MAX)` — changing the on-chain selector that the dApp had requested. This is a footgun for users.
3. The on-chain allowance was never queried, so the UI could not show the *resulting* total — only the delta, which is easy to misread as an absolute approve.

## Design Decisions
- **Re-encode with the original selector.** When the user edits the delta, the vault re-encodes using the same selector the dApp originally sent (`increaseAllowance` stays `increaseAllowance`). The dApp's intent is preserved; we never silently upgrade to `approve(MAX)`.
- **Hide the Unlimited toggle for increase variants.** Forcing users through a separate code path rather than silently flipping the selector. Keeps the editor honest about what will be sent on chain.
- **Drop `decreaseAllowance` from the editable set.** The earlier feat commit included it, but on review there's no UX win — users rarely need to *increase* a decrease — and supporting it widens the surface area without value. Scoped down in the follow-up refactor.
- **Backfill `approveType`/`spender`/`amountParsed` from the local decoder** onto server-returned Approve display components so dApp-initiated flows stay in sync after the user edits the delta. Without this, server-side rendering would override the locally-edited values.
- **Track consumed local approves via a `Set`** in `ServiceSignatureConfirm` so the positional fallback can't double-consume an entry already matched by token — fixes a misattribution bug across multi-approve txs.
- **Render `'Infinite'` / non-finite `amountParsed` as unlimited wording** instead of `NaN` (`increaseAllowance(MaxUint256)` leaks `'Infinite'` through the parsed amount path).
- **Prefix finite increase deltas with `+`** so they can't be misread as an absolute approve while the current-allowance lookup is still loading.
- **Route caught errors through `defaultLogger.app.error.log`** in `useTokenApproveAllowance` instead of silently swallowing — debuggability for the otherwise invisible allowance failures.

## Changes Detail
- `packages/kit-bg/src/vaults/impls/evm/decoder/abi.ts` + `Vault.ts` — add inc/dec selectors to the approve action classification; preserve original selector on re-encode.
- `packages/kit-bg/src/services/ServiceSignatureConfirm.ts` — set-based dedupe for local approve matching; backfill `approveType`/`spender`/`amountParsed` onto server-returned components.
- `packages/kit/src/views/ApproveEditor/ApproveEditor.tsx` + `packages/kit/src/views/SignatureConfirm/components/ApproveEditor/ApproveEditor.tsx` — Current/Final allowance rows; hide Unlimited toggle for increase variants; new i18n keys (`approve_edit_increase_amount`, `approve_edit_current_allowance`, `approve_edit_final_allowance`).
- `packages/kit/src/components/TxAction/TxActionTokenApprove.tsx` + `Assets.tsx` — render `+delta` for finite increases; unlimited wording for non-finite; include `increaseAllowance(MAX)` in editable-approve gate.
- `packages/kit/src/hooks/useTokenApproveAllowance.ts` (new) — fetch current on-chain allowance via `/swap/v1/allowance`; log errors instead of swallowing.
- `packages/shared/types/{tx,signatureConfirm,swap/types}.ts` + `txActionUtils.ts` — type plumbing for the new approveType variants and post-tx total.
- Locale JSON updates for the three new i18n keys across all supported languages.

## Risk Assessment
- **Risk Level**: Medium
- **Affected Platforms**: Desktop / Mobile / Web / Extension (signature confirm flow is shared)
- **Risk Areas**:
  - **Selector preservation** during re-encode is the highest-stakes change — a bug here would silently rewrite the on-chain call. Manual verification on each variant is required.
  - **Cross-token attribution** in `ServiceSignatureConfirm` for multi-approve transactions — the set-based dedupe is the fix, but multi-approve txs are an edge case worth eyeballing.
  - **Allowance lookup failures** are now logged but still non-fatal — the editor falls back to delta-only display, which is the prior behavior, so no regression risk.

## Test plan
- [ ] dApp triggers `increaseAllowance(spender, delta)` — editor opens, shows current allowance, final = current + delta, Unlimited toggle hidden.
- [ ] dApp triggers legacy `increaseApproval(spender, delta)` (selector `0xd73dd623`) — same as above.
- [ ] dApp triggers `increaseAllowance(spender, MaxUint256)` — UI renders unlimited wording (no NaN), still editable.
- [ ] User edits delta in the editor — confirm tx is re-encoded with the *original* selector (verify via decoded calldata, not approve(...)).
- [ ] dApp triggers a multi-approve tx mixing two tokens — each row shows the correct token's delta (no cross-attribution).
- [ ] Allowance lookup fails (simulate by blocking `/swap/v1/allowance`) — error appears in `defaultLogger.app.error`, editor still functions with delta-only display.
- [ ] Verify Chinese / Japanese / Korean locales render the three new strings correctly.

---

<a href="https://app.devin.ai/review/onekeyhq/app-monorepo/pull/11487" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

### CODE REVIEW

**Comment by revan-zhang** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scan Engine |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/382d9e2a-9784-487a-b13b-1d83108e4911?refs=be134302c9d93d55e6a208972d1e4e67fa5aa7a1&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/382d9e2a-9784-487a-b13b-1d83108e4911/license?refs=be134302c9d93d55e6a208972d1e4e67fa5aa7a1&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

File: packages/kit-bg/src/vaults/impls/evm/Vault.ts [hunk 1]
```diff
@@ -983,14 +998,23 @@ export default class Vault extends VaultBase {
...
+        tokenApproveInfo.approveType ??
+        action.tokenApprove.approveType ??
+        EApproveType.Approve;
+
+      const selector = SELECTOR_BY_APPROVE_TYPE[approveType];
+      // Unlimited only applies to absolute approve; increase variants always
+      // encode the value as a delta.
+      const isAbsoluteUnlimited =
+        approveType === EApproveType.Approve && isUnlimited;
       const amountHex = toBigIntHex(
```
Comment (originalix) [inline packages/kit-bg/src/vaults/impls/evm/Vault.ts:1016]: 当原始 calldata 是 `increaseAllowance`/`increaseApproval` 且参数为 `MaxUint256` 时，本地解码会把 amount 显示成 `Infinite`。编辑器的 Reset/还原路径会把这个 `Infinite` 原样作为 `tokenApproveInfo.allowance` 传回这里，但当前分支只把 `isUnlimited` 应用于绝对 `approve`，increase 分支会执行 `new BigNumber('Infinite').shiftedBy(decimals)`，最终生成 `0xNaN` 并在 `defaultAbiCoder.encode` 处抛错。

这会让用户在编辑/重置这类授权时无法恢复原交易，属于签名前 calldata 重写路径的运行时断点。建议在 `_updateTokenApproveInfo` 里统一识别 `InfiniteAmountText`/非 finite allowance，并在任何 `approveType` 下都映射为 `MaxUint256` 后再编码；`isUnlimited` 不应是唯一能编码 `MaxUint256` 的入口。

File: packages/kit/src/hooks/useTokenApproveAllowance.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    async () => {
+      if (!enabled || !spender || !tokenAddress) return null;
+      const walletAddress =
+        await backgroundApiProxy.serviceAccount.getAccountAddressForApi({
+          accountId,
+          networkId,
+        });
+      if (!walletAddress) return null;
+      try {
+        const r = await backgroundApiProxy.serviceSwap.fetchApproveAllowance({
```
Comment (originalix) [inline packages/kit/src/hooks/useTokenApproveAllowance.ts]: 这个 hook 是只为展示 current/final allowance 的被动查询，但它复用了 `serviceSwap.fetchApproveAllowance`。该 service 在每次请求前都会调用全局 `cancelCheckTokenApproveAllowance()`，因此任意签名确认行、编辑器或多个 increase approval 组件同时挂载时，后一个展示查询会取消前一个 allowance 检查。

影响是当前/最终 allowance 预览可能退化成 `-` 或 delta-only；同一个 service 还被 Swap/BulkSend 的授权状态机使用，外部取消会打断它们原本假设“只有自己的下一次请求才会取消上一次”的 loading/error 收敛。建议给预览场景新增一个不触发全局 cancel 的 read-only allowance 方法，并让这个 hook 使用它；交互型 swap 检查继续保留原来的取消语义。

File: packages/kit-bg/src/vaults/impls/evm/Vault.ts (hunk 1 above)
Comment (weatherstar) [reply packages/kit-bg/src/vaults/impls/evm/Vault.ts:1016]: Fixed in 09470c02ce. `_updateTokenApproveInfo` now treats `InfiniteAmountText`/non-finite allowance as `MaxUint256` regardless of `approveType`, so the editor/reset path no longer feeds `Infinite` into `BigNumber(...).shiftedBy(decimals)` on the increase branches. The previous absolute-approve `isUnlimited` short-circuit is folded into the same check.

File: packages/kit/src/hooks/useTokenApproveAllowance.ts (hunk 2 above)
Comment (weatherstar) [reply packages/kit/src/hooks/useTokenApproveAllowance.ts]: Fixed in 09470c02ce. Added `serviceSwap.fetchApproveAllowanceForDisplay`, a read-only allowance lookup that does not call `cancelCheckTokenApproveAllowance()` and does not share the abort controller with the swap/bulk-send authorization flows. `useTokenApproveAllowance` now uses it, so passive previews no longer cancel each other or the interactive swap state machines. The original `fetchApproveAllowance` keeps its existing cancel semantics for those interactive call sites.

**Review by zhaono1 — DISMISSED** [review]

File: packages/shared/src/utils/txActionUtils.ts [hunk 3]
```diff
@@ -462,6 +462,8 @@ function convertTokenApproveActionToSignatureConfirmComponent({
     isEditable: !isRevoke && !isMultiTxs,
     isInfiniteAmount: action.isInfiniteAmount,
     networkId,
+    approveType: action.approveType,
```
Comment (originalix) [inline packages/shared/src/utils/txActionUtils.ts:471]: 这里已经把 `action.approveType` 传给 Approve component，但上面的 `isRevoke` 仍然只用 `amount === 0` 判断。

触发 `increaseAllowance(spender, 0)` 或 `increaseApproval(spender, 0)` 时，本质是增量授权 0，而不是 revoke；当前机制会把它归类成撤销，进而影响多交易标题、spender 行文案和 `isEditable`。签名前展示会把 calldata 语义说反。

建议把 `isRevoke` 收紧为 `approveType` 是绝对 `approve` 且金额为 0，或者按 `approveType` 分开生成 revoke / increase 的标签和可编辑状态。

File: packages/kit/src/views/SignatureConfirm/components/SignatureConfirmComponents/Assets.tsx [hunk 4]
```diff
@@ -346,17 +367,38 @@ function AssetsTokenApproval(props: IAssetsApproveProps) {
     [editable, component.isEditable],
   );
 
+  const displayedAmount = useMemo(() => {
+    const amountBN = new BigNumber(component.amountParsed);
+    // increaseAllowance(MaxUint256) leaks the "Infinite" sentinel through
+    // amountParsed (isInfiniteAmount is only set on absolute approve), so
+    // collapse both signals into one unlimited check.
+    if (component.isInfiniteAmount || !amountBN.isFinite()) {
```
Comment (originalix) [inline packages/kit/src/views/SignatureConfirm/components/SignatureConfirmComponents/Assets.tsx]: 这里把 `!amountBN.isFinite()` 和 `isInfiniteAmount` 合并成“无限授权”，但对 `increaseAllowance` / `increaseApproval` 来说，`MaxUint256` 是“增加 MaxUint256”，不是绝对设置为无限额度。

触发 `increaseAllowance(spender, MaxUint256)` 且当前 allowance 非 0 时，实际 calldata 可能因为加法溢出而 revert；即使不 revert，用户看到的“无限授权”也没有表达增量语义。签名前展示会弱化用户对实际调用参数的判断。

建议在 `isIncrease` 分支里单独处理非有限 delta，显示为增量语义，不要复用绝对 `approve` 的 unlimited 展示。

File: packages/kit/src/components/TxAction/TxActionTokenApprove.tsx [hunk 5]
```diff
@@ -108,6 +121,8 @@ function TxActionTokenApproveListView(props: ITxActionProps) {
           symbol: approveSymbol,
         },
       );
+    } else if (isIncrease) {
+      title = `Increase ${approveSymbol} allowance`;
```
Comment (originalix) [inline packages/kit/src/components/TxAction/TxActionTokenApprove.tsx]: 这里新增了用户可见的英文字符串。

触发非英文语言环境，或者 current allowance 还没取回走到 fallback 文案时，签名确认和历史详情会绕过 `ETranslations`，导致英文混杂。

建议为 increase allowance 的标题和 fallback 说明补充 `ETranslations` 文案，并在这里通过 `intl.formatMessage` 渲染。

File: packages/shared/src/utils/txActionUtils.ts (hunk 3 above)
Comment (weatherstar) [reply packages/shared/src/utils/txActionUtils.ts:471]: Fixed in be893b6ed6. `convertTokenApproveActionToSignatureConfirmComponent` (and the multi-tx title path in `getSwapHistoryDescription`) now require `approveType` to be absolute `EApproveType.Approve` (or undefined for legacy chains) before treating amount=0 as revoke. `increaseAllowance(spender, 0)` / `increaseApproval(spender, 0)` keep approve semantics — multi-tx label, spender row copy, and `isEditable` all stay on the increase branch.

File: packages/kit/src/views/SignatureConfirm/components/SignatureConfirmComponents/Assets.tsx (hunk 4 above)
Comment (weatherstar) [reply packages/kit/src/views/SignatureConfirm/components/SignatureConfirmComponents/Assets.tsx]: Fixed in be893b6ed6. The increase branch in `displayedAmount` now runs before the absolute-unlimited check, so `increaseAllowance(MaxUint256)` no longer collapses into the absolute "Unlimited" copy. Non-finite deltas render as `+Unlimited` (with finite `currentAllowance` summed when available), and `finalAllowanceParsed` returns null for non-finite deltas instead of silently falling back to `currentAllowanceParsed`, which previously hid the increment from users.

File: packages/shared/src/locale/enum/translations.ts [hunk 6]
```diff
@@ -147,6 +147,8 @@ export enum ETranslations {
   approve_edit_approve_amount = 'approve_edit.approve_amount',
   approve_edit_current_allowance = 'approve_edit.current_allowance',
   approve_edit_final_allowance = 'approve_edit.final_allowance',
+  approve_edit_increase_allowance = 'approve_edit.increase_allowance',
+  approve_edit_increase_allowance_by_amount = 'approve_edit.increase_allowance_by_amount',
```
[the remark this answers is not in the document]
Comment (weatherstar) [reply packages/shared/src/locale/enum/translations.ts:151]: Not a violation in practice. The repo workflow regularly commits `translations.ts` and the locale JSON files via the i18n pipeline (see recent `chore: update locale` commits like `baf30fea00`, `fd7680a693` directly touching these files). The "auto-generated" rule means do not hand-craft locale content; the generated output IS committed and must be in sync with consumers like `ETranslations`. The new keys here were produced through that pipeline.

File: packages/shared/types/swap/types.ts [hunk 7]
```diff
@@ -701,7 +701,8 @@ export interface ISwapApproveAllowanceResponse {
   isApproved: boolean;
   allowanceTarget: string;
   shouldApproveAmount: string;
-  approvedAmount: string;
+  // sic — backend spelling. Value is already decimal-parsed, not raw wei.
+  approveAmounted: string;
```
[the remark this answers is not in the document]
Comment (weatherstar) [reply packages/shared/types/swap/types.ts:705]: Fixed in CLI mirror. The shared type rename (`approvedAmount` → `approveAmounted`) reflects backend reality — the backend has always sent the typo `approveAmounted`. The CLI mirror at `apps/cli/src/commands/swap/swap-execute.ts:57` was the side that was wrong (silently reading `undefined`). Updated the CLI interface to match the shared type and added the same `// sic — backend spelling` comment for consistency. No read sites in `apps/cli/src/` reference the field, so this is a type-only fix.

**Review by originalix — APPROVED** [review]
